### ISSUE TITLE
SelFlow: Self-Supervised Learning of Optical Flow

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#105]
*CVPR 2019*

# Motivation

- Previous unsupervised optical flow estimations fail to handle diverse types of natural occlusions successfully.

# Key Idea

- Randomly inject super-pixel level noise and train OCC model on pseudo labels from Non-OCC model.
- Expand to multiple frame setting. (3 or more)

# Previous Approach

- DDFlow: feed randomly cropped image patch to the student model with pseudo labels from teacher model to reason about occlusion

# Method

![image](https://user-images.githubusercontent.com/46308407/129998752-07ee10e2-7557-42df-8021-a7a853c6335a.png)

- Overall framework is similar to DDFlow.
- Instead of random crop, employ SILC based super-pixel noise injection.
- NOC model is trained without artificial occlusions, and OCC model reasons about noise (occluded regions) with signals from NOC model.
- PWC-Net baseline as in DDFlow.
![image](https://user-images.githubusercontent.com/46308407/129998860-1787fd7b-438a-4f1b-aeeb-4740e6bd7884.png)
- Expand to multiple frame setting for more accurate optical flow estimation.
![image](https://user-images.githubusercontent.com/46308407/129998891-8d6bc263-3c0c-48f2-9280-3057724ae76f.png)
- Forward Backward consistency based occlusion mask
![image](https://user-images.githubusercontent.com/46308407/129998934-fcfbb103-1b50-4313-834b-94d835b89938.png)

## Objective
- Only photometric loss for NOC model.
![image](https://user-images.githubusercontent.com/46308407/129998996-60ff5915-248e-4180-ba79-644d81bc9a69.png)
- Photometric loss as well as occlusion loss for OCC model.
![image](https://user-images.githubusercontent.com/46308407/129999035-cd6d7d67-30d2-40b6-844c-8a0f5791373a.png)
![image](https://user-images.githubusercontent.com/46308407/129999026-90703657-5ba9-40a5-b2ce-03b5deff4567.png)
![image](https://user-images.githubusercontent.com/46308407/129999045-6e63d9a2-13c3-4268-b25e-bec62c49ee64.png)

## Supervised Fine-tuning
![image](https://user-images.githubusercontent.com/46308407/129999071-c4aafb59-5227-4b97-b911-3b0a128a0653.png)
- Great performance when init. with SelFlow and further finetune with labels.

# Experiments

## Baseline
![image](https://user-images.githubusercontent.com/46308407/129999137-9178a0bb-8aca-4f51-a013-5b8ed4822ddc.png)

## Ablation
![image](https://user-images.githubusercontent.com/46308407/129999210-26f058f2-c4d2-4a40-976b-e13f27050706.png)
- Note that switching from rectangle to superpixel yields significant performance boost, especially for occluded regions.

## Supervised Finetuning
![image](https://user-images.githubusercontent.com/46308407/129999280-6c1bb46f-bbc4-4473-8066-9dd22c357d45.png)

# Comments

- Same authors from DDFlow submitted another, very similar yet improved paper shortly after.
- Generalizing rectangular random crop to super-pixel base noise injection was a natural idea.
- Describing their method in terms of SSL must've been a way to argue for novelty. They changed network names from student-teacher to NOC-OCC for novelty sake, presumably speaking.
- Improve performance with 3 frame setting. It's quite obvious that feeding another frame would give better results. Not sure if this is indeed a noteworthy contribution.
